transport-native-unix-common/src/main/c/netty_unix_errors.c

```diff
@@ -29,12 +29,59 @@ static jclass portUnreachableExceptionClass = NULL;
 static jclass closedChannelExceptionClass = NULL;
 static jmethodID closedChannelExceptionMethodId = NULL;
 
+/**
+ Our `strerror_r` wrapper makes sure that the function is XSI compliant,
+ even on platforms where the GNU variant is exposed.
+ Note: `strerrbuf` must be initialized to all zeros prior to calling this function.
+ XSI or GNU functions do not have such a requirement, but our wrappers do.
+ */
+#if (_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && ! _GNU_SOURCE
+    static inline int strerror_r_xsi(int errnum, char *strerrbuf, size_t buflen) {
+        return strerror_r(errnum, strerrbuf, buflen);
+    }
+#else
+    static inline int strerror_r_xsi(int errnum, char *strerrbuf, size_t buflen) {
+        char* tmp = strerror_r(errnum, strerrbuf, buflen);
```

Comment:
do we need to check for `tmp == NULL` ? 

https://linux.die.net/man/3/strerror_r

Reply:
I've look at both https://linux.die.net/man/3/strerror_r and https://man7.org/linux/man-pages/man3/strerror.3.html and I don't see a case where the GNU specific `strerror_r` would return `NULL`. It either allocates a new string, or references a static string.

Reply:
ok